src/style-spec/types.ts

```diff
@@ -392,6 +392,13 @@ export type ModelSourceSpecification = {
     "tiles"?: Array<string>
 }
 
+export type CanvasSourceSpecification = {
```

Comment:
`CanvasSource` is purely a GL JS runtime entity. Thus, it doesn't belong to the [Style Specification](https://docs.mapbox.com/style-spec/reference/sources/). You can alternatively import the type from `canvas_source.ts` in [`src/source/source_types.ts`](https://github.com/mapbox/mapbox-gl-js/blob/main/src/source/source_types.ts) it is a bit will change convention, but it reflects actual state anyway